src/testing/jest/jest-apis.ts

```diff
@@ -0,0 +1,48 @@
+/*!
+ * This file contains Jest API usages for situations where it is difficult to determine which API should be used.
+ *
+ * An example of this is determining the version of Jest, which is retrieved via the `getVersion` API.
+ * It's difficult at compile & runtime to determine:
+ * 1. If such an API exists
+ * 2. If it's typings are the same across all versions of Jest
+ * 3. If there are variants of this API, which one to use and when
+ *
+ * Short of probing the directory where a user keeps their modules (e.g. `node_modules/`), we need to make a "best
+ * guess" at things. This file is meant to only contain functions for these types of scenarios. It is expected that this
+ * file be added to sparingly.
+ */
+
+import { getVersion } from 'jest';
+
+// TODO(STENCIL-959): Improve this typing by narrowing it
+export type JestPuppeteerEnvironment = any;
+
+type Jest26CacheKeyOptions = { instrument: boolean; rootDir: string };
+type Jest26Config = { instrument: boolean; rootDir: string };
+type Jest27TransformOptions = { config: Jest26Config };
+export type JestPreprocessor = {
```

Comment:
Another spot where it'd be nice if the abstract layer didn't need to worry about version-specific APIs. Anything we can do with generics here? How are these types consumed?

Reply:
> Another spot where it'd be nice if the abstract layer didn't need to worry about version-specific APIs. 

Trust me, I wish we could 😆 The problem is that the generic version of our connectors/adapters need to return _something_. And those something's need to be compatible with Jest.

> Anything we can do with generics here? How are these types consumed?

Right now, they're only used by `jest-adapter.ts`. I think in the near future, we'll end up refactoring this to be less aware of these types. But the exact shape/form hasn't "presented itself" yet - I have a feeling once we get Jest 28 in place, that'll start to become more evident